src/pymor/models/saddle_point.py

```diff
@@ -0,0 +1,124 @@
+# This file is part of the pyMOR project (https://www.pymor.org).
+# Copyright pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (https://opensource.org/licenses/BSD-2-Clause)
+
+from pymor.models.basic import StationaryModel
+from pymor.operators.block import BlockColumnOperator, BlockOperator
+from pymor.operators.constructions import AdjointOperator, VectorOperator
+from pymor.operators.interface import Operator
+from pymor.vectorarrays.block import BlockVectorArray
+from pymor.vectorarrays.interface import VectorArray
+
+
+class StationarySaddelPointModel(StationaryModel):
+    r"""Generic class for Saddel point models.
+
+    Constructs a stationary model for a linear saddle-point problems
+    using a 2x2 |BlockOperator|:
+
+    .. math::
+        \begin{bmatrix}
+            A & B^\ast \\
+            B & C
+        \\end{bmatrix}
+        \begin{bmatrix}
+            u \\ p
+        \\end{bmatrix}
+        =
+        \begin{bmatrix}
+            f \\ g
+        \\end{bmatrix},
+
+    Here, ``A`` is square on the ``u``-space, ``B`` maps from the ``u``-space
+    to the ``p``-space, and ``C`` is optional (stabilization/mass) on the``p``-space.
+    Typical example: Stokes equation with velocity (``u``) and pressure (``p``).
+
+    Parameters
+    ----------
+    A
+        |Operator| on the ``u``-space with ``A.source == A.range``.
+    B
+        Coupling |Operator| between ``u`` and ``p``-space with ``B.source == A.source``.
+    C
+        Optional mass |Operator| on the ``p``-space. One of:
+        - |Operator| with ``C.source == C.range == B.range``.
+        - ``None``, which yields a |ZeroOperator|.
+    rhs
+        Right-hand side. Several forms are supported and coerced to a
+        |BlockColumnOperator| of shape ``(2, 1)``:
+        - |BlockColumnOperator| with blocks ``[f, g]``.
+        - |BlockVectorArray| with blocks ``[f, g]``.
+        - |VectorOperator| on the ``u``-space. Then ``g`` is set to zero.
+        - |VectorArray| on the ``u``-space. Then ``g`` is set to zero.
+
+    products
```

Comment:
Since only two keys are allowed, I think `u_product` and `p_product` parameters would make more sense?